### PR TITLE
HTBHF-1390 A query limits to message retrieval repository method.

### PR DESCRIPTION
**Pull request by mikeSmith08**

### CODE REVIEW

File: api/src/main/resources/application.yml [hunk 1]
```diff
@@ -57,6 +57,7 @@ message-processor:
   # for an explanation of the ISO-8601 duration format
   minimum-lock-time: "PT1M"
   maximum-lock-time: "PT60M"
+  message-limit: 100
```
Comment (YetAnotherMatt) [inline api/src/main/resources/application.yml]: I think we may want to make this a bit bigger than 100. Shall we try 1000 and see how fast we can process them? We probably want about as many as we can process in a minute.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java [hunk 2]
```diff
@@ -44,8 +48,8 @@ public void processAllMessages() {
     }
 
     private void processMessagesOfType(MessageType messageType) {
-        List<Message> messages = messageRepository.findAllMessagesByTypeOrderedByDate(messageType);
-        if (CollectionUtils.isEmpty(messages)) {
+        List<Message> messages = messageRepository.findAllMessagesByTypeOrderedByDate(messageType, PageRequest.of(0, messageProcessingLimit));
+        if (messages.isEmpty()) {
```
Comment (dwybourn) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java]: Do we no longer have to deal with a null response?

File: api/src/main/resources/application.yml (hunk 1 above)
Comment (mikeSmith08) [reply api/src/main/resources/application.yml]: Changed to 1000.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java (hunk 2 above)
Comment (mikeSmith08) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java]: Yes, we do, I've changed it back.

**Pull request merged**